ndcube/ndcube.py

```diff
@@ -264,57 +267,22 @@ def missing_axis(self):
     def pixel_to_world(self, *quantity_axis_list):
         # The docstring is defined in NDDataBase
 
-        origin = 0
-        list_arg = []
-        indexed_not_as_one = []
-        result = []
-        quantity_index = 0
-        for i in range(len(self.missing_axes)):
-            wcs_index = self.wcs.naxis-1-i
-            # the cases where the wcs dimension was made 1 and the missing_axes is True
-            if self.missing_axes[wcs_index]:
-                list_arg.append(self.wcs.wcs.crpix[wcs_index]-1+origin)
-            else:
-                # else it is not the case where the dimension of wcs is 1.
-                list_arg.append(quantity_axis_list[quantity_index].to(u.pix).value)
-                quantity_index += 1
-                # appending all the indexes to be returned in the answer
-                indexed_not_as_one.append(wcs_index)
-        list_arguments = list_arg[::-1]
-        pixel_to_world = self.wcs.all_pix2world(*list_arguments, origin)
-        # collecting all the needed answer in this list.
-        for index in indexed_not_as_one[::-1]:
-            result.append(u.Quantity(pixel_to_world[index], unit=self.wcs.wcs.cunit[index]))
-        return result[::-1]
+        quantity_axis_list = quantity_axis_list[::-1]
+        pixel_to_world = self.wcs.pixel_to_world(*quantity_axis_list)
+        return pixel_to_world[::-1]
 
     def world_to_pixel(self, *quantity_axis_list):
         # The docstring is defined in NDDataBase
+        
+        result = list()
+        world_to_pixel = self.wcs.world_to_pixel(*quantity_axis_list)
```

Comment:
I believe that just like ```pixel_to_world``` above, the ```quantity_list``` must be reversed before being put into ```self.wcs.world_to_pixel``` and then the output reversed again before being returned to the user.

Reply:
Done.